backend/icpy/services/chat_service.py

```
@@ -389,14 +422,192 @@ def _ensure_session_meta(self, session_id: str, name: Optional[str] = None) -> N
+            except Exception as e:
+                logger.debug(f"[AUTOTITLE] Provider {provider_name} failed: {e}")
+                continue
+
+        logger.info("[AUTOTITLE] All providers failed, returning None for heuristic fallback")
+        return None
+
+    def _schedule_auto_title(self, session_id: str) -> None:
+        """Schedule auto title generation (one per session)."""
+        logger.info(f"[AUTOTITLE] _schedule_auto_title called for {session_id}")
+        if not self._should_auto_title(session_id):
+            logger.info(f"[AUTOTITLE] _schedule_auto_title: skipped (not eligible)")
```

Ruff (0.15.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)